diff -up a/config.def.h b/config.def.h
--- a/config.def.h	2021-04-28 18:40:05.715904305 +0200
+++ b/config.def.h	2021-04-28 18:38:18.115907934 +0200
@@ -251,6 +251,7 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
 	{ ShiftMask,            XK_Page_Up,     kscrollup,      {.i = -1} },
 	{ ShiftMask,            XK_Page_Down,   kscrolldown,    {.i = -1} },
+	{ TERMMOD,              XK_Return,      newterm,        {.i =  0} },
 };
 
 /*
diff -up a/st.c b/st.c
--- a/st.c	2021-04-28 18:40:05.669237641 +0200
+++ b/st.c	2021-04-28 18:37:32.962576121 +0200
@@ -160,6 +160,7 @@ typedef struct {
 } STREscape;
 
 static void execsh(char *, char **);
+static char *getcwd_by_pid(pid_t pid);
 static void stty(char **);
 static void sigchld(int);
 static void ttywriteraw(const char *, size_t);
@@ -1062,6 +1063,26 @@ int tisaltscr(void)
 }
 
+void
+newterm(const Arg* a)
+{
+	switch (fork()) {
+	case -1:
+		die("fork failed: %s\n", strerror(errno));
+		break;
+	case 0:
+		chdir(getcwd_by_pid(pid));
+		execlp("st", "./st", NULL);
+		break;
+	}
+}
+
+static char *getcwd_by_pid(pid_t pid) {
+	char buf[32];
+	snprintf(buf, sizeof buf, "/proc/%d/cwd", pid);
+	return realpath(buf, NULL);
+}
+
 void
 tswapscreen(void)
 {
 	Line *tmp = term.line;
diff -up a/st.h b/st.h
--- a/st.h	2021-04-28 18:40:05.669237641 +0200
+++ b/st.h	2021-04-28 18:39:06.659239634 +0200
@@ -84,6 +84,7 @@ void redraw(void);
 void tfulldirt(void);
 void draw(void);
 
+void newterm(const Arg *);
 void kscrolldown(const Arg *);
 void kscrollup(const Arg *);
 void printscreen(const Arg *);
